feat(docs): Improve documentation and enforce markdown linting (#16252)
island-is/island.is: libs/application/templates/social-insurance-administration/income-plan/README.md

> ## Description

Reviewer: The header has been removed.
```suggestion
# Application Template Income Plan

## Description
```

This application template facilitates the submission of an income plan post-application.

### States

#### Prerequisites

> This temporary state is where all new income plans are initially created. It is unlisted and has a short lifespan. It acts as a gatekeeper to the actual income plan, fetching data from Þjóðskrá and TR. Applicants cannot proceed if a previous income plan has been in progress at TR for less than 10 days.

Reviewer: A requirement has been added here.
```suggestion
This temporary state is where all new income plans are initially created. It is unlisted and has a short lifespan. It acts as a gatekeeper to the actual income plan, fetching data from Þjóðskrá and TR. Applicants cannot proceed if a previous income plan has been in progress at TR for less than 10 days or if they do not have an active application at TR that requires an income plan.
```

# Income Plan

## Description

This application template enables the submission of an income plan post-application.

### States

#### Prerequisites

> All new income plans are initially in this brief, unlisted state, fetching data from Þjóðskrá and TR. Applicants must wait 10 days if a previous income plan is active at TR.

Author: @ylfahfa, is this correct? Your last comment got overwritten :speak_no_evil: 
```suggestion
This temporary state is where all new income plans are initially created. It is unlisted and has a short lifespan. It acts as a gatekeeper to the actual income plan, fetching data from Þjóðskrá and TR. Applicants cannot proceed if a previous income plan has been in progress at TR for less than 10 days or if they do not have an active application at TR that requires an income plan.
```

Reviewer: I just committed the suggested change
